apps/functions-subscription/src/domain/trial.ts

```diff
@@ -15,12 +18,20 @@ interface TrialIdBrand {
   readonly TrialId: unique symbol;
 }
 export const TrialIdCodec = t.brand(
-  NonEmptyString,
-  (str): str is t.Branded<NonEmptyString, TrialIdBrand> => str.length > 0,
+  WithinRangeString(26, 27),
+  (str): str is t.Branded<WithinRangeString<26, 27>, TrialIdBrand> =>
+    str.length >= 26 && str.length < 27,
```

Comment:
**question**: why this change?

it this change necessary? is it necessary on the domain level?

Reply:
creating the response object on `apps/functions-subscription/src/adapters/azure/functions/codec.ts` requires to cast a trialId (`NonEmptyString`) to nextId/previousId (`WithinRangeString<26, 27>`).
this could be done in three ways:
- by casting the trialId to unknown before casting it to NonEmptyString
- by changing the trialId type to match nextId/previousId
- by removing the minLength and maxLength properties from nextId/previousId

I went for the second options since `WithinRangeString<26, 27>` matches the properties for the trialId type too and it avoids calls to the api with invalid trialIds.
this also required me to update the open api specification and the trialIds in the tests to match the 26 characters requirement.

Reply:
I get it, but here you are updating a value on the domain and I don't think we need to be that stricter there.

If within the adapter we need to handle a different type, then I'd manipulate the `TrialId` there, so we decouple the domain to the adapters.

Reply:
Resolved in eff99bbaf41355bc1a98ee131ab9a4d2694136f4